src/Features/CSharp/Portable/QuickInfo/CSharpDiagnosticAnalyzerQuickInfoProvider.cs

```diff
@@ -0,0 +1,145 @@
+                IdentifierNameSyntax { Parent: PragmaWarningDirectiveTriviaSyntax directive } identifier when directive.IsDisablePragma()
+                    => (directive, identifier),
+                _ => default,
+            };
+
+            if (errorCode != null)
+            {
+                // First look in the analyzer diagnostics of the document. By doing so we get detailed information about the actual
+                // diagnostic message and the code that is affected by the diagnostic id.
+                // If this fails, try to find the error code in the SupportedDiagnostics of all referenced analyzers.
+                return
+                    (await GetQuickInfoFromDiagnosticAnalyzerAsync(document, pragmaWarning, errorCode, cancellationToken).ConfigureAwait(false))
+                    ?? GetQuickInfoFromSupportedDiagnosticsOfProjectAnalyzers(document, errorCode);
+            }
+
+            return null;
+        }
+
+        private async Task<QuickInfoItem?> GetQuickInfoFromDiagnosticAnalyzerAsync(
+            Document document,
+            PragmaWarningDirectiveTriviaSyntax pragmaWarning,
+            IdentifierNameSyntax errorCode,
+            CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (root != null)
+            {
+                var range = TextSpan.FromBounds(pragmaWarning.FullSpan.End, root.FullSpan.End);
+                var diagnostics = await _diagnosticAnalyzerService.GetDiagnosticsForSpanAsync(document, range,
+                    diagnosticIdOpt: errorCode.Identifier.ValueText,
+                    includeSuppressedDiagnostics: true, cancellationToken: cancellationToken).ConfigureAwait(false);
+                var supressedDiagnostic = diagnostics.OrderBy(d => d.DataLocation?.SourceSpan?.Start ?? int.MaxValue).FirstOrDefault();
+                if (supressedDiagnostic != null)
+                {
+                    var relatedSpans = supressedDiagnostic.HasTextSpan
+                        ? new[] { supressedDiagnostic.GetTextSpan() }
+                        : Array.Empty<TextSpan>();
+                    return CreateQuickInfo(errorCode,
+                        supressedDiagnostic.Message ?? supressedDiagnostic.Title ?? supressedDiagnostic.Id,
+                        relatedSpans);
+                }
+            }
+
+            return null;
+        }
+
+        private static QuickInfoItem? GetQuickInfoFromSupportedDiagnosticsOfProjectAnalyzers(Document document,
+            IdentifierNameSyntax errorCode)
+        {
+            var analyzerReferences = document.Project.AnalyzerReferences.Union(document.Project.Solution.AnalyzerReferences);
+            var supportedDiagnostics = from r in analyzerReferences
+                                       from a in r.GetAnalyzersForAllLanguages()
+                                       from d in a.SupportedDiagnostics
```

Comment:
`SupportedDiagnostics` can throw any exception as it is custom extension code, we should not invoke it directly here.